File: backend/core-service/src/Fix.Application/UseCases/Mandates/Commands/IssueMandateCommand.cs
using Fix.Application.Abstractions.Authorization;
using Fix.Application.Abstractions.Context;
using Fix.Application.Abstractions.Messaging;
using Fix.Application.Mandates.Dtos;
using Fix.Domain.AggregateRoots.Mandates;
using Fix.Domain.AggregateRoots.Roles;

namespace Fix.Application.Mandates.Commands;

[RequireRole(RoleCodes.CreateMandate)]
public sealed record IssueMandateCommand(
    Guid UserId,
    Guid OrganizationId,
    Guid PolicyId,
    Guid AxisId,
    MandateType Type,
    MandateTermsInput Terms)
    : ICommand<MandateDto>, IOrganizationRequest;

File: frontend/bff/src/application/mandates/commands/issue-mandate.command.ts
import type { MandateTermsInput } from './mandate-terms.input.js';
import type { MandateType } from '../../../cross-cutting/enums/mandate-type.js';
import type { RequestContext } from '../../../cross-cutting/context/request-context.js';

export interface IssueMandateCommand {
  context: RequestContext;
  policyId: string;
  axisId: string;
  type: MandateType;
  terms: MandateTermsInput;
}
